risks/MASVS-STORAGE/1-store-sensitive-data-securely/data-unencrypted-external/android-data-unencrypted-external-frida/example-1/run.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# SUMMARY: This script uses frida-trace to trace files that an app has opened since it spawned
+# The script filters the output of frida-trace to print only the paths belonging to external
+# storage but the the predefined list of external storage paths might not be complete.
+# A sample output is shown in "output.txt". If the output is empty, it indicates that no external
+# storage is used.
+
+frida-trace \
+    -U \
+    -f com.example \
+    -i "open" \
```

Comment:
Hooking `open` will miss the files created using the `MediaStore` API because it seems to typically leverage pre-existing file descriptors or obtain them through complex interactions with content providers, which manage the actual file operations on behalf of the app.

If we'd hook `open` and `write`, when a `MediaStore` API operation occurs we'd only see `write` and not to `open`.

Reply:
Do you think hooking `contentResolver.insert(MediaStore.Downloads.EXTERNAL_CONTENT_URI, contentValues)` will be enough to cover MediaStore? Or should we also hook `OutputStream`? Or maybe get a file path by FD from `write`?